hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerKeyMapperTask.java

```diff
@@ -383,4 +398,60 @@ private void handlePutOMKeyEvent(String key, OmKeyInfo omKeyInfo,
     }
   }
 
+  /**
+   * Write an OM key to container DB and update containerID -> no. of keys
+   * count to the Global Stats table.
+   *
+   * @param key key String
+   * @param omKeyInfo omKeyInfo value
+   * @param containerKeyCountMap we keep the containerKey counts in this map 
+   *                             to allow batching to containerKeyCountTable
+   *                             after reprocessing is done
+   * @throws IOException if unable to write to recon DB.
+   */
+  private void handleKeyReprocess(String key,
+                                  OmKeyInfo omKeyInfo,
+                                  Map<Long, Long> containerKeyCountMap)
+      throws IOException {
+    long containerCountToIncrement = 0;
+    for (OmKeyLocationInfoGroup omKeyLocationInfoGroup : omKeyInfo
+        .getKeyLocationVersions()) {
+      long keyVersion = omKeyLocationInfoGroup.getVersion();
+      for (OmKeyLocationInfo omKeyLocationInfo : omKeyLocationInfoGroup
+          .getLocationList()) {
+        long containerId = omKeyLocationInfo.getContainerID();
+        ContainerKeyPrefix containerKeyPrefix = new ContainerKeyPrefix(
+            containerId, key, keyVersion);
+        if (reconContainerMetadataManager.getCountForContainerKeyPrefix(
+            containerKeyPrefix) == 0) {
+          // Save on writes. No need to save same container-key prefix
+          // mapping again.
+          reconContainerMetadataManager.storeContainerKeyMapping(
```

Comment:
Thanks for the patch @ivandika3 , though I have one suggestion. After this change, we are storing containerKeyMapping info with every key which is hitting DB everytime, so if we want to improve memory footprint due to containerKeyMap, we can use a threshold and till that threshold number of keys, retain in memory, lets say 1000 keys, and then use batchStore function for containerKeyMap instead of removing it completely. We have done like this in other aync tasks like NSSummary, You can refer that. This way we'll avoid 1 billion DB calls with 1 billion keys.

Reply:
@devmadhuu Thank you for the review. I have added configuration and support for incremental batching for containerKeyMapping. Please take another look.

Reply:
@devmadhuu The threshold might be too low, could you help suggest a good default? Or maybe Ishould follow the NSSummary configuration (i.e. 150,000)?

Reply:
@ivandika3 @devmadhuu what was the conclusion here?

Reply:
thanks @ivandika3 for the testing and updating the patch!